lib/DBIx/SearchBuilder/Handle/mysql.pm

```diff
@@ -372,4 +372,10 @@ sub HasSupportForCombineSearchAndCount {
     }
 }
 
+sub CastAsDecimal {
+    my $self  = shift;
+    my $field = shift or return;
+    return "($field+0)";
```

Comment:
Can you add a comment here to explain why we aren't using CAST for mysql and mariadb?

Reply:
updated accordingly and created PR 8.